### PR TITLE
fix: Add EIP1559 fee parameters

### PR DESCRIPTION
**Pull request by rafinskipg**
Types are not being generated with the new EIP1559 fees, I hope this fix is correct. 🤕

### CODE REVIEW

**Comment by krzkaczor** [comment]
Hey, thanks for this PR. I assume that support for eip 1559 landed in ethers already. Probably we should try to reuse types exposed by ethers rather than redefining them... i don't remember what was the reasoning for this. Does eip1559 work in ethers-v4 too?

**Comment by rafinskipg** [comment]
@krzkaczor  I didn't try ethers, i just realized that the new types were not being generated locally so i just added them. Types were added in web3-js 1.5.1 but i don't know about ethers

File: packages/target-truffle-v4/static/types.d.ts [hunk 1]
```diff
@@ -40,6 +40,8 @@ declare namespace Truffle {
     from?: string
     gas?: BigNumber | number | string
     gasPrice?: BigNumber | number | string
+    maxPriorityFeePerGas?: string | number | BN
+    maxFeePerGas?: string | number | BN
```
Comment (anthanh) [inline packages/target-truffle-v4/static/types.d.ts:44]:
```suggestion
    maxFeePerGas?: BigNumber | number | string
```

File: packages/target-truffle-v4/static/types.d.ts [hunk 2]
```diff
@@ -40,6 +40,8 @@ declare namespace Truffle {
     from?: string
     gas?: BigNumber | number | string
     gasPrice?: BigNumber | number | string
+    maxPriorityFeePerGas?: string | number | BN
```
Comment (anthanh) [inline packages/target-truffle-v4/static/types.d.ts:43]:
```suggestion
    maxPriorityFeePerGas?: BigNumber | number | string
```

File: packages/target-truffle-v5-test/types/truffle-contracts/types.d.ts [hunk 3]
```diff
@@ -62,6 +62,8 @@ declare namespace Truffle {
     from?: string;
     gas?: BN | number | string;
     gasPrice?: BN | number | string;
+    maxPriorityFeePerGas?: string | number | BN;
+    maxFeePerGas?: string | number | BN;
```
Comment (anthanh) [inline packages/target-truffle-v5-test/types/truffle-contracts/types.d.ts:66]:
```suggestion
    maxFeePerGas?: BN | number | string;
```
Same order like others.

File: packages/target-truffle-v5-test/types/truffle-contracts/types.d.ts [hunk 4]
```diff
@@ -62,6 +62,8 @@ declare namespace Truffle {
     from?: string;
     gas?: BN | number | string;
     gasPrice?: BN | number | string;
+    maxPriorityFeePerGas?: string | number | BN;
```
Comment (anthanh) [inline packages/target-truffle-v5-test/types/truffle-contracts/types.d.ts:65]:
```suggestion
    maxPriorityFeePerGas?: BN | number | string;
```
Same order like others

**Review by anthanh — COMMENTED** [review]
Actually, looks like using ethers exported types, but checks fails at truffle-v4 because of a missing type.

File: packages/target-truffle-v5-test/types/truffle-contracts/types.d.ts (hunk 4 above)
Comment (anthanh) [reply packages/target-truffle-v5-test/types/truffle-contracts/types.d.ts:65]: wdyt @krzkaczor ?

**Comment by krzkaczor** [comment]
Thanks for working on this. I will make sure to merge this during the weekend. It probably needs some runtime tests as well...

**Comment by krzkaczor** [comment]
New PR: https://github.com/ethereum-ts/TypeChain/pull/456#pullrequestreview-736480312 (h/t @zemse)

Thanks for the initial implementation! In the WIP PR we try to rely on ethers exported types instead of redeclaring them. It should be merged soon.

**Pull request closed without merging**